### PR TITLE
Add support for description on instances

### PR DESCRIPTION
**Pull request by cmoesel**
Instances can now use the Description keyword.  The description will be put into the ImplementationGuide JSON and rendered in the documentation.

Other changes:
- the InstanceDefinition now carries metadata in _instanceMeta
- the ImplementationGuide will use an instance's Title as its name if it is set

### CODE REVIEW

**Comment by cmoesel** [comment]
NOTE: I just tacked on a commit to update a couple of log messages, as requested in #214.

File: src/ig/IGExporter.ts [hunk 1]
```diff
@@ -294,9 +294,10 @@ export class IGExporter {
     examples.forEach(example => {
```
Comment (ngfreiter) [inline src/ig/IGExporter.ts]: Thank you GitHub for not letting me comment on the line I want, but 293 is 
`const examples = sortBy(this.pkg.instances, instance => instance.id ?? instance.instanceName);`
This should be changed to use `instance._instanceMeta.instanceName` right?

File: src/fhirtypes/InstanceDefinition.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import cloneDeep = require('lodash/cloneDeep');
 import { Meta } from './specialTypes';
 import { HasId } from './common';
 import { applyMixins } from '../utils';
+import { isNull } from 'util';
```
Comment (ngfreiter) [inline src/fhirtypes/InstanceDefinition.ts]: As GitHub has pointed out, 'isNull' is defined but never used.

File: src/ig/IGExporter.ts (hunk 1 above)
Comment (cmoesel) [reply src/ig/IGExporter.ts]: Yes.  Not sure how that one escaped my global search!

File: src/fhirtypes/InstanceDefinition.ts (hunk 2 above)
Comment (cmoesel) [reply src/fhirtypes/InstanceDefinition.ts]: Ah, dang.  I remember that happening.  VS Code auto-completed something to `isNull`, which I did not intend -- so I fixed it, but hadn't realized that VS Code also auto-added an import for it.  Doh.

**Comment by cmoesel** [comment]
@ngfreiter -- fixed!

**Pull request merged**